crates/router/src/connector/forte/transformers.rs

```diff
@@ -63,62 +66,93 @@ impl TryFrom<utils::CardIssuer> for ForteCardType {
-                let action = match item.request.is_auto_capture()? {
+    fn try_from(
+        (_currency_unit, _currency, amount, item): (
+            &types::api::CurrencyUnit,
+            types::storage::enums::Currency,
+            i64,
+            T,
+        ),
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            amount,
+            router_data: item,
+        })
+    }
+}
+impl
+    TryFrom<
+        &ForteRouterData<
+            &types::RouterData<
+                types::api::payments::Authorize,
+                PaymentsAuthorizeData,
+                PaymentsResponseData,
+            >,
+        >,
+    > for FortePaymentsRequest
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &ForteRouterData<
+            &types::RouterData<
+                types::api::payments::Authorize,
+                PaymentsAuthorizeData,
+                PaymentsResponseData,
+            >,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let payment_data = match &item.router_data.request.payment_method_data {
+            api::PaymentMethodData::Card(card) => {
+                let action = match item.router_data.request.is_auto_capture()? {
                     true => ForteAction::Sale,
                     false => ForteAction::Authorize,
                 };
-                let card_type = ForteCardType::try_from(ccard.get_card_issuer()?)?;
-                let address = item.get_billing_address()?;
+                let card_type = ForteCardType::try_from(card.get_card_issuer()?)?;
+                let address = item.router_data.get_billing_address()?;
                 let card = Card {
                     card_type,
-                    name_on_card: ccard.card_holder_name.clone(),
-                    account_number: ccard.card_number.clone(),
-                    expire_month: ccard.card_exp_month.clone(),
-                    expire_year: ccard.card_exp_year.clone(),
-                    card_verification_value: ccard.card_cvc.clone(),
+                    name_on_card: card.card_holder_name.clone(),
+                    account_number: card.card_number.clone(),
+                    expire_month: card.card_exp_month.clone(),
+                    expire_year: card.card_exp_year.clone(),
+                    card_verification_value: card.card_cvc.clone(),
                 };
                 let billing_address = BillingAddress {
                     first_name: address.get_first_name()?.to_owned(),
                     last_name: address.get_last_name()?.to_owned(),
                 };
-                let authorization_amount =
-                    utils::to_currency_base_unit_asf64(item.request.amount, item.request.currency)?;
+                let authorization_amount = utils::to_currency_base_unit_asf64(
+                    item.router_data.request.amount,
+                    item.router_data.request.currency,
+                )?;
```

Comment:
You have to remove this function call to `to_currency_base_unit_asf64`, that is the task of this PR.


Please refer to the PR in your tagged issue, on how to do it and update accordingly

Reply:
After updating the `amount` as `f64` in `ForteRouterData` you can directly use `let authorisation_amount = item.amount;`